src/mmap.cpp

```diff
@@ -49,22 +49,22 @@ mmap_reads(off_t partial_file_size,
            int32_t &offset_pair_from_top,
```

Comment:
since there was a change in type on line 52, should this also be uint? 

is mmap support limited to 2**32 files?